common/kafka/kafka_config.cpp

```diff
@@ -0,0 +1,63 @@
+/// Copyright 2019 Pinterest Inc.
+///
+/// Licensed under the Apache License, Version 2.0 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+/// http://www.apache.org/licenses/LICENSE-2.0
+
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+
+#include <fstream>
+
+#include "glog/logging.h"
+#include "common/kafka/kafka_config.h"
+
+namespace kafka {
+
+/**
+ * Read Java kafka client configuration file
+ */
+bool KafkaConfig::read_conf_file(
+  const std::string &conf_file,
+  ConfigMap *configMap) {
+  std::ifstream inf(conf_file.c_str());
+
+  if (!inf) {
+    LOG(ERROR) << ": " << conf_file << ": could not open file" << std::endl;
+    return false;
+  }
+
+  LOG(INFO) << ": " << conf_file << ": read config file" << std::endl;
+
+  std::string line;
+  int linenr = 0;
+
+  while (std::getline(inf, line)) {
+    linenr++;
+
+    // Ignore comments and empty lines
+    if (line.length() == 0 || line[0] == '#')
+      continue;
+
+    // Match on key=value..
+    size_t d = line.find("=");
+    if (d == 0 || d == std::string::npos) {
+      LOG(INFO) << ": " << conf_file << ":" << linenr << ": " << line
+                << ": invalid line (expect key=value): " << ::std::endl;
+      return false;
+    }
+
+    std::string key = line.substr(0, d);
+    std::string val = line.substr(d + 1);
+
+    (*configMap)[key] = std::move(val);
+  }
+  inf.close();
```

Comment:
Can you close the file under a SCOPE_EXIT so that it closes whenever the function scope ends? In the current case, it doesn't close the file when returning on line 52

You can follow this example: https://github.com/pinterest/rocksplicator/blob/ec91db9af275648ea7905eea33409be6af61f1c3/common/availability_zone.cpp#L55

Reply:
Done.